src/codegen/cfg.rs

```diff
@@ -657,10 +657,10 @@ impl ControlFlowGraph {
                 dimensions, values, ..
             } => format!(
                 "constant {} [ {} ]",
-                dimensions
-                    .iter()
-                    .map(|d| format!("[{d}]"))
-                    .collect::<String>(),
+                dimensions.iter().fold(String::new(), |mut output, d| {
+                    let _ = write!(output, "[{d}]");
```

Comment:
is `let _  = ` necessary?

Reply:
It is:
```
warning: unused `std::result::Result` that must be used
   --> src/codegen/cfg.rs:661:21
    |
661 |                     write!(output, "[{d}]");
    |                     ^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: this `Result` may be an `Err` variant, which should be handled
    = note: `#[warn(unused_must_use)]` on by default
    = note: this warning originates in the macro `write` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Reply:
So your `let _ = ` swallows errors, that's not great. It should be use `.unwrap()` or `.expect()`.

Reply:
The `let _ = ` was clippy's suggestion. https://rust-lang.github.io/rust-clippy/master/index.html#/format_collect
I'll change it to unwrap.